drivers/sensor/stmemsc/stmemsc.h

```diff
@@ -76,5 +107,36 @@ int stmemsc_spi_read(const struct spi_dt_spec *stmemsc,
 		     uint8_t reg_addr, uint8_t *value, uint8_t len);
 int stmemsc_spi_write(const struct spi_dt_spec *stmemsc,
 		      uint8_t reg_addr, uint8_t *value, uint8_t len);
+
+/*
+ * Populate the stmdev_ctx_t structure pointed by stmdev_ctx_ptr with
+ * specific stmemsc sp APIs that set reg_addr bit6 to '1' in order to allow
+ * multiple read/write operations. This is common in some STMEMSC drivers
+ */
+#define STMEMSC_CTX_SPI_INCR(stmdev_ctx_ptr)				\
+	.ctx = {							\
+		.read_reg = (stmdev_read_ptr) stmemsc_spi_read_incr,	\
+		.write_reg = (stmdev_write_ptr) stmemsc_spi_write_incr,	\
+		.mdelay = (stmdev_mdelay_ptr) stmemsc_mdelay,		\
+		.handle = (void *)stmdev_ctx_ptr			\
+	}
+
+int stmemsc_spi_read_incr(const struct spi_dt_spec *stmemsc,
+			  uint8_t reg_addr, uint8_t *value, uint8_t len);
+int stmemsc_spi_write_incr(const struct spi_dt_spec *stmemsc,
+			   uint8_t reg_addr, uint8_t *value, uint8_t len);
+
+/*
+ * Populate the stmdev_ctx_t structure pointed by stmdev_ctx_ptr with
+ * custom stmemsc spi APIs specified by driver.
+ */
+#define STMEMSC_CTX_SPI_CUSTOM(stmdev_ctx_ptr, spi_rd_api, spi_wr_api)	\
+	.ctx = {							\
+		.read_reg = (stmdev_read_ptr) spi_rd_api,		\
+		.write_reg = (stmdev_write_ptr) spi_wr_api,		\
+		.mdelay = (stmdev_mdelay_ptr) stmemsc_mdelay,		\
+		.handle = (void *)stmdev_ctx_ptr			\
```

Comment:
can you use a consistent style for the castings? (no space after the `)` is what I normally see). Also are the casting really needed?

Reply:
ok, space removed. Yes, I think casting is required:

```
typedef int32_t (*stmdev_write_ptr)(void *, uint8_t, const uint8_t *, uint16_t);                                                                                                                          
typedef int32_t (*stmdev_read_ptr)(void *, uint8_t, uint8_t *, uint16_t);
typedef void (*stmdev_mdelay_ptr)(uint32_t millisec);
```